### DISCUSSION TITLE
Read File activity

### DISCUSSION THREAD

**Discussion by nilesh-darji1** [elsa-workflows/elsa-core#2868] [Q&A]
I am new to elsa workflow and i am trying to read Textfile with ReadFile activity and i am unable to  convert it back to text using Javascript. Is there a activity or some solution to achieve the conversation.

**Comment by sfmskywalker** [comment]
Hi @nilesh-darji1, you can use the `String.fromCharCode` JavaScript function to convert the bytes returned by the ReadFile activity into a string.

For example. let's say you have a workflow with two activities:

1. ReadFile with a name of ReadFile1
2. WriteLine to print the contents of some text file that was read by ReadFile1.

The WriteLine's Text field could be configured using the following JavaScript:

```javascript
const bytes = activities.ReadFile1.Bytes(); // Get the bytes stored in the ReadFile1 activity.
const text = String.fromCharCode(...bytes); // Convert the bytes to a string.
return text; // Return the string to write to the console.
```

Here is the workflow in action:

![read-file-workflow](https://user-images.githubusercontent.com/938393/159699021-260e807e-32d9-455c-93aa-8df2daf5d5ce.gif)
